lib/devise_two_factor/models/two_factor_authenticatable.rb

```diff
@@ -78,7 +78,7 @@ module ClassMethods
                                     :otp_secret_encryption_key)
 
         def generate_otp_secret(otp_secret_length = self.otp_secret_length)
-          ROTP::Base32.random_base32(otp_secret_length)
+          ROTP::Base32.random(otp_secret_length * 5/8)
```

Comment:
Change since rotp 5.0:

ROTP::Base32.random_base32 is now ROTP::Base32.random and the argument has changed from secret string length to byte length to allow for more precision. There is an alias to allow for random_base32 for the time being.